lib/rspec/core/reporter.rb

```diff
@@ -79,6 +88,17 @@ def message(message)
       notify :message, Notifications::MessageNotification.new(message)
     end
 
+    # @param event [Symbol] Name of the custom event to trigger on formatters
+    # @param options [Hash] Hash of arguments to provide via `CustomNotification`
+    #
+    # Publish a custom event to supporting registered formatters.
+    # @see RSpec::Core::Notifications::CustomNotification
+    def publish(event, options={})
```

Comment:
Do you think it's worth naming this `publish_custom_event` (or similar) to make it clear that this is only for custom events.